juriscraper/pacer/acms_attachment_page.py

```diff
@@ -81,6 +83,54 @@ def check_validity(self, parsed_json: dict) -> None:
             return
         self.is_valid = True
 
+    def query(self, case_id: str, entry_id: str):
+        """
+        Retrieves details for a specific docket entry and its attachments.
+
+        This method first fetches all docket entries for a case, then isolates
+        the target entry by its ID, and finally retrieves any associated
+        attachments.
+
+        :param case_id: The unique identifier of the case.
+        :param entry_id: The unique identifier of the specific docket entry to
+                         retrieve.
+        """
+        # Fetch Docket Entry Details
+        docket_info = self.api_client.get_docket_entries(case_id)
+
+        # Find the specific docket entry by its ID.
+        # If the entry is not found, 'entry_data' will be None.
+        entry_data = next(
+            filter(
+                lambda v: v["docketEntryId"] == entry_id,
+                docket_info["docketEntries"],
+            ),
+            None,
+        )
+        if not entry_data:
+            logger.warning(
+                f"Docket entry with ID '{entry_id}' not found in case '{case_id}'."
+            )
+            self.is_valid = False
+            self._acms_json = {}
+
```

Comment:
Isn't an early return required here? Otherwise, if there is no `entry_data`, the lookup `entry_data["restrictedPartyFilingDocketEntry"]` will fail.

Reply:
Thanks for pointing this out! I'll add an early return.